### PR TITLE
Numpy frontend - random poisson

### LINKED ISSUE #9767: poisson
Add Random sampling to Numpy Frontend #1530

### PR DESCRIPTION
**Pull request by gladysmawarni**
Close #9767

### CODE REVIEW

**Comment by guillesanbri** [comment]
Hi! Thank you very much for the contribution :grin:

The test seems to fail with the PyTorch backend (`ValueError: only one element tensors can be converted to Python scalars`), maybe this change helps :slightly_smiling_face:

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py [hunk 1]
```diff
@@ -152,3 +152,45 @@ def test_numpy_normal(
...
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        lam=x,
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:193]: given that it's only one array, we can probably extract it changing this to `lam=x[0]`

Comment (gladysmawarni) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py]: Thank you for the correction! Just fixed the mistake, please let me know if there's anything else I need to do :)

**Comment by guillesanbri** [comment]
Looks good to me, the test seems to fail in the CI but it's passing for me locally. I'll check if it's something on our side which probably is and update soon, thanks once again for the contribution! :grin:

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py [hunk 2]
```diff
@@ -136,6 +136,46 @@ def test_numpy_normal(
...
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        lam=x[0],
+        test_values=False,
+
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:177]: missing a `)` here after the master merge :slightly_smiling_face:

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py [hunk 3]
```diff
@@ -136,6 +136,46 @@ def test_numpy_normal(
...
+    fn_tree,
+    on_device,
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:171]: these four lines have to be replaced with `test_flags=test_flags`

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py [hunk 4]
```diff
@@ -136,6 +136,46 @@ def test_numpy_normal(
...
+    size=st.tuples(
+        st.integers(min_value=1, max_value=10), st.integers(min_value=2, max_value=2)
+    ),
+)
+def test_numpy_poisson(
+    dtype_and_x,
+    size,
+    as_variable,
+    num_positional_args,
+    native_array,
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:160]: these four lines have to be replaced with `test_flags,`

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py [hunk 5]
```diff
@@ -136,6 +136,46 @@ def test_numpy_normal(
...
+        shape=st.tuples(
+            st.integers(min_value=1, max_value=2)
+        ),
+        min_value=1,
+        max_value=100
+    ),
+
+    size=st.tuples(
+        st.integers(min_value=1, max_value=10), st.integers(min_value=2, max_value=2)
+    ),
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py:153]: seems like this is not used so we can remove it :slightly_smiling_face:

**Review by guillesanbri — CHANGES_REQUESTED** [review]
Indeed, there had been a refactor on our side for the decorator, after you merged master the error now shows correctly :smile: 

I have added a few comment to remove some unused code and instructions to use the new helper, once these are done I'll merge it, thanks!

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py (hunk 5 above)
Comment (guillesanbri) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py]: ahh, I see what happened sorry :sweat_smile: when you merged master `size` was moved to the function below along with the missing parenthesis, you should indeed add this size param, but also pass it to the test, sorry for the confusion!

Comment (gladysmawarni) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_random/test_functions.py]: no worries, I was not paying attention when merging 😅. Just added back the `size` argument, thank you again for the review!

**Comment by guillesanbri** [comment]
Looks good to me! Thank you very much for contributing! :rocket:

**Pull request merged**